presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PickTableLayout.java

```diff
@@ -250,9 +278,48 @@ public static PlanNode pushPredicateIntoTableScan(
             SqlParser parser,
             ExpressionDomainTranslator domainTranslator)
     {
+        if (metadata.isPushdownFilterSupported(session, node.getTable())) {
+            Map<NodeRef<Expression>, Type> predicateTypes = getExpressionTypes(
+                    session,
+                    metadata,
+                    parser,
+                    types,
+                    predicate,
+                    emptyList(),
+                    WarningCollector.NOOP,
+                    false);
+
+            BiMap<VariableReferenceExpression, VariableReferenceExpression> symbolToColumnMapping = node.getAssignments().entrySet().stream()
```

Comment:
Curious: Why do we need `BiMap` here? Looks like `RowExpressionNodeInliner.replaceExpression` requires an `Map` as last parameter?

Reply:
@wenleix The inverted mapping is used to convert unenforcedConstraint into an expression that is used to create a FilterNode on top of the TableScanNode. The call to `.inverse()` got lost in the last round of changes and the test didn't catch it because Hive connector never returns unenforcedConstraint that's not TRUE. I added the missing call.

```
            RowExpression unenforcedConstraint = pushdownFilterResult.getUnenforcedConstraint();
            if (!TRUE.equals(unenforcedConstraint)) {
                return new FilterNode(idAllocator.getNextId(), tableScan, replaceExpression(unenforcedConstraint, symbolToColumnMapping.inverse()));
            }
```